mlflow/store/sqlalchemy_store.py

```diff
@@ -227,11 +228,14 @@ def get_metric(self, run_uuid, metric_key):
         return metric.value
 
     def search_runs(self, experiment_ids, search_expressions, run_view_type):
-        raise NotImplementedError()
-
-    def list_run_infos(self, experiment_id, _=None):
-        exp = self.session.query(SqlExperiment).get(experiment_id)
-        infos = []
-        for run in exp.runs:
-            infos.append(run.to_mlflow_entity().info)
-        return infos
+        runs = [self._get_run(r.run_uuid, run_view_type).to_mlflow_entity()
+                for exp in experiment_ids
+                for r in self.list_run_infos(exp, run_view_type)]
+        if len(search_expressions) == 0:
+            return runs
+        return [r for r in runs if all([does_run_match_clause(r, s) for s in search_expressions])]
```

Comment:
`all([])` is True -- do we need the if statement before?

Reply:
`len(search_expressions)` should take care of legit empty search expression cases. We should not see any other scenarios of empty list sent to `all`.

If a future bug gets introduced in `does_run_match_clause` whereby one of the branches exits without a return value the function will return a `None` and `all` handles that well.